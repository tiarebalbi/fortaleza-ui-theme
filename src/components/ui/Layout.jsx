// @flow

import * as React from 'react';
import styled, { ThemeProvider } from 'styled-components';

import type { ContentType, LayoutType } from '../../types';

import { theme, media } from '../../themes';

const ContentStyle = styled.div`
  left: 0;
  margin-left: 110px;
  padding-top: 65px;
  position: relative;

  .details {
    margin-left: 280px;
    margin-top: 20px;
    margin-right: 30px;
  }

  ${media.tablet`
     margin-left: 90px;
     
    .details { 
      margin-left: 210px;
    }
  `};
`;

export const Content = (props: ContentType) => (
  <ContentStyle>{props.children}</ContentStyle>
);

const MobileBlocker = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  padding-top: 30px;
  text-align: center;
  font-size: 30px;
  z-index: -5000;
  visibility: hidden;
  display: none;
  background: #333;
  color: #fff;

  ${media.phone`
    z-index: 5000;
    visibility: visible;
    display: block;
  `};
`;

const Wrapper = styled.div`
  background: #f2f9ff;
  height: 100%;
`;

export const Layout = (props: LayoutType) => (
  <ThemeProvider theme={theme}>
    <Wrapper>
      <MobileBlocker>
        Este sistema n√£o suporta acesso via celular.
      </MobileBlocker>
      {props.children}
    </Wrapper>
  </ThemeProvider>
);
