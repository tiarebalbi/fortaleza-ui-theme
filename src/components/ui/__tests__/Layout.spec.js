import React from 'react';
import { Content, Layout, Details } from '../Layout';
import { mount } from 'enzyme';

describe('Layout', () => {
  describe('Content', () => {
    const wrapper = mount(<Content>Sample</Content>);

    it('should be stateless', () => {
      expect(wrapper.instance()).toBeNull();
    });

    it('should be able to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('Details', () => {
    const wrapper = mount(<Details>Sample</Details>);

    it('should be stateless', () => {
      expect(wrapper.instance()).toBeNull();
    });

    it('should be able to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should be able to find by class', () => {
      expect(wrapper.find('.details')).toExist();
    });
  });
  describe('Layout', () => {
    const wrapper = mount(<Layout>Sample</Layout>);

    it('should be stateless', () => {
      expect(wrapper.instance()).toBeNull();
    });

    it('should be able to match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
