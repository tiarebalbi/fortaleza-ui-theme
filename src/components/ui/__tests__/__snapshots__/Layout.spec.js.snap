// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Content should be able to match snapshot 1`] = `
<Content>
  <styled.div>
    <div
      className="sc-bdVaJa JoqXo"
    >
      Sample
    </div>
  </styled.div>
</Content>
`;

exports[`Layout Layout should be able to match snapshot 1`] = `
<Layout>
  <ThemeProvider
    theme={
      Object {
        "fonts": Object {
          "pre": "Consolas, Liberation Mono, Courier, monospace",
          "primary": "Roboto, sans-serif",
          "quote": "Georgia, serif",
        },
        "palette": Object {
          "alert": Array [
            "#FFD166",
            "#CBA238",
            "#997600",
            "#694D00",
            "#412600",
          ],
          "danger": Array [
            "#EF476F",
            "#CC1F54",
            "#A9003A",
            "#870023",
            "#650009",
          ],
          "grayscale": Array [
            "#38383D",
            "#6F707A",
            "#CACDDF",
            "#D3D6E4",
            "#EBECF3",
          ],
          "primary": Array [
            "#194B86",
            "#456CAA",
            "#6C8ED0",
            "#92B2F7",
            "#B8D8FF",
          ],
          "secondary": Array [
            "#FF9E00",
            "#D07800",
            "#A35300",
            "#793000",
            "#550A00",
          ],
          "success": Array [
            "#06D6A0",
            "#00AB78",
            "#008153",
            "#005930",
            "#00340F",
          ],
          "white": Array [
            "#fff",
            "#F9F9FC",
            "#919191",
            "#5E5E5E",
            "#303030",
          ],
        },
        "reversePalette": Object {
          "alert": Array [
            "#412600",
            "#694D00",
            "#997600",
            "#CBA238",
            "#FFD166",
          ],
          "danger": Array [
            "#650009",
            "#870023",
            "#A9003A",
            "#CC1F54",
            "#EF476F",
          ],
          "grayscale": Array [
            "#EBECF3",
            "#D3D6E4",
            "#CACDDF",
            "#6F707A",
            "#38383D",
          ],
          "primary": Array [
            "#B8D8FF",
            "#92B2F7",
            "#6C8ED0",
            "#456CAA",
            "#194B86",
          ],
          "secondary": Array [
            "#550A00",
            "#793000",
            "#A35300",
            "#D07800",
            "#FF9E00",
          ],
          "success": Array [
            "#00340F",
            "#005930",
            "#008153",
            "#00AB78",
            "#06D6A0",
          ],
          "white": Array [
            "#303030",
            "#5E5E5E",
            "#919191",
            "#F9F9FC",
            "#fff",
          ],
        },
        "sizes": Object {
          "maxWidth": "1100px",
        },
      }
    }
  >
    <styled.div>
      <div
        className="sc-htpNat kzWjOO"
      >
        <styled.div>
          <div
            className="sc-bwzfXH fKyror"
          >
            Este sistema n√£o suporta acesso via celular.
          </div>
        </styled.div>
        Sample
      </div>
    </styled.div>
  </ThemeProvider>
</Layout>
`;
